test/Core.ts

```diff
@@ -384,45 +386,110 @@ export class Foo {
+
+        const stringifyPackageName = ({ scope, name }: PackageName): string =>
+          `${O.fold(
+            () => '',
+            (str) => `@${str}/`
+          )(scope)}${name}`
+
+        const stringifyPackageImport = ({ quotes, packageName, lib, path }: PackageImport) => {
+          const quote = quotes === 'single' ? "'" : '"'
+          const libStr = lib ? '/lib' : ''
+          const pathStr = O.fold(
+            () => '',
+            (str) => `/${str}`
+          )(path)
+          const packageNameStr = stringifyPackageName(packageName)
+          return `${quote}${packageNameStr}${libStr}${pathStr}${quote}`
+        }
+
+        const mkState = (packageName: PackageName, importLine: string) =>
+          prefixWithCwd({
+            'package.json': JSON.stringify({ name: stringifyPackageName(packageName), homepage: 'http://foo' }),
+            'src/utils/foo.ts': `
 /**
  * @since 0.0.1
  */
-
 /**
  * @category utils
  * @since 0.0.1
  */
 export class Foo {
   /**
    * @example
-   * import * as Foo from "${JSON.parse(defaultPackageJson)['name']}"
+   * ${importLine}
+   * 
+   * const bar = 1
    *
    * @since 0.0.1
    */
   public bar(): string {
     return 'bar'
   }
-}
-          `
-        })
-        const capabilities = makeCapabilities(state)
-
-        assertRight(await Core.main(capabilities)(), (result) => {
-          const exampleFileName = `${[...process.cwd().split('/'), 'src', 'utils', 'foo.ts'].join(
-            '-'
-          )}-Foo-method-bar-0.ts`
-          const exampleFilePath = `${process.cwd()}/docs/examples/${exampleFileName}`
-          const importFromRegex = new RegExp("from '../../src'")
+                            }`
+          })
+
+        /* By using the Array monad we generate a couple of PackageImport permutations
+         */
+        const packageImportPermutations: PackageImport[] = pipe(
+          A.bindTo('quotes')(['single' as const, 'double' as const]),
+          A.bind('scope', () => [O.some('my-org'), O.none]),
+          A.bind('name', () => ['my-package']),
+          A.bind('packageName', () =>
+            pipe(
+              A.bindTo('scope')([O.some('my-org'), O.none]),
+              A.bind('name', () => ['my-package'])
+            )
+          ),
+          A.bind('lib', () => [true, false]),
+          A.bind('path', () => [O.some('seg1'), O.some('seg1/seg2'), O.none])
+        )
 
-          assert.strictEqual(result, undefined)
-          assert.strictEqual(command, 'ts-node')
-          assert.strictEqual(executablePath.includes('docs/examples/index.ts'), true)
-          assert.strictEqual(importFromRegex.test(fileSystemState[exampleFilePath]), true)
-        })
+        test.each(
+          pipe(
+            packageImportPermutations,
+            A.map((packageImport) => [stringifyPackageImport(packageImport), packageImport])
+          )
+        )(
+          'should replace imports like %s from the project with the local source folder',
```

Comment:
That's a bit unfortunate in Jest's implementation of `test.each`. The description argument has to be a string with those formatting placeholders. As we want to do a bit more preprocessing, we have to map the stringified version beforehand onto a tuple.

Reply:
But generally it works nice, it produces error messages as if each test was written inside an `it`.

Reply:
I think this is fine if it gives us clean error messages. This way if something fails in CI we can quickly troubleshoot it.